### PR TITLE
Fix page and missing return on unadopted repos API

### PR DESCRIPTION
**Pull request by qwerty287**
Page must be 1 if it's not specified and it should return after sending an internal server error.

### CODE REVIEW

**Review by zeripath — APPROVED** [review]

File: routers/api/v1/admin/adopt.go [hunk 1]
```diff
@@ -43,9 +43,13 @@ func ListUnadoptedRepositories(ctx *context.APIContext) {
 	//     "$ref": "#/responses/forbidden"
 
 	listOptions := utils.GetListOptions(ctx)
+	if listOptions.Page <= 0 {
```
Comment (lunny) [inline routers/api/v1/admin/adopt.go]: Maybe it should be zero, then we can ignore page to use `-1`.

Comment (qwerty287) [reply routers/api/v1/admin/adopt.go]: Sorry, I don't understand your point. Why should the page be -1? if it's <= 0, it should be 1 usually, or do you want to change that it's 1 if it is <= 0?

Comment (lunny) [reply routers/api/v1/admin/adopt.go]: I mean how to ignore the pagination?

Comment (qwerty287) [reply routers/api/v1/admin/adopt.go]: Why should it ignore pagination, it's a list endpoint and should provide pagination like every other list endpoint?

Comment (qwerty287) [reply routers/api/v1/admin/adopt.go]: https://github.com/go-gitea/gitea/pull/18848/commits/95793162e00d40985dee516d59ba43adb8ed4706

**Review by lunny — APPROVED** [review]

**Comment by lunny** [comment]
make L-G-T-M work

**Pull request merged**